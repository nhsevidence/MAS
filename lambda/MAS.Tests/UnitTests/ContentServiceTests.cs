using MAS.Configuration;
using MAS.Services;
using MAS.Tests.Infrastructure;
using Shouldly;
using System.Linq;
using System.Threading.Tasks;
using Xunit;

namespace MAS.Tests.UnitTests
{
    public class ContentServiceTests : TestBase
    {
        [Fact]
        public async Task ReadSingleItem()
        {
            //Arrange
            AppSettings.CMSConfig = TestAppSettings.GetSingleItemFeed();
            var contentService = new ContentService();

            //Act
            var result = await contentService.GetItemAsync("");

            //Assert
            result.Id.ShouldBe("5daeb5af22565a82530d7373"); 
            result.Title.ShouldBe("Wonder drug");
        }

        [Fact]
        public async Task ReadMultipleItems()
        {
            //Arrange
            AppSettings.CMSConfig = TestAppSettings.GetMultipleItemsFeed();
            var contentService = new ContentService();

            //Act
            var result = await contentService.GetItemsAsync();

            //Assert
            result.Count().ShouldBe(4);
            result.FirstOrDefault().Id.ShouldBe("5daf1aa18a34d4bb8405b5e0");
            result.FirstOrDefault().Title.ShouldBe("Wonder Drug");
            result.LastOrDefault().Id.ShouldBe("5db6cbcc8a34d4ca5905b5e4");
            result.LastOrDefault().Title.ShouldBe("A Placebo");
        }
    }
}
