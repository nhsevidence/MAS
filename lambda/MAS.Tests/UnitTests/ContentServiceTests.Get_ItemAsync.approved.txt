{
    "_id":"5dc30790b01357811811c43d",
    "Title":"some title",
    "Source":"5dc306e9b01357d0f311be50",
    "ShortSummary":"Short summary"
}